File: components/sandbox/scripts/genericClient.py
#Ice.loadSlice("--all slice/generic.ice")
import Ice
Ice.loadSlice("--all -I../common/target/generated-sources/ -I../common/target/generated-sources/slice slice/generic.ice")
from main.eis.generic import *

comm = Ice.initialize()

# Create a proxy for the invoice factory object.
proxy = comm.stringToProxy("server:tcp -p 10000")

# Narrow the proxy to the proper type.
generic = ServerPrx.checkedCast(proxy)

generic.query(None)

comm.shutdown()

File: components/sandbox/src/main/ice/GenericClient.java
package main.ice;

import ome.model.core.ImageRemote;

import main.eis.generic.ServerPrx;
import main.eis.generic.ServerPrxHelper;

public class GenericClient extends Ice.Application
{

    public static void main(String[] args)
    {
        GenericClient app = new GenericClient();
        int status = app.main("Client", args, "config");
        System.exit(status);
    }

    public int run(String[] args)
    {

        ServerPrx engine = ServerPrxHelper.checkedCast(communicator()
                .stringToProxy("server:tcp -p 10000"));
        if (engine == null)
        {
            System.err.println("invalid engine proxy");
            return 1;
        }

        ImageRemote[] ir = engine.query("null");

        // communicator().waitForShutdown();
        communicator().shutdown();

        return 0;
    }

}
